### ISSUE TITLE
Deleting content from PMP crashes player

### ISSUE THREAD

**Issue by cedell** [plexinc/plex-media-player#313]
PMP 2.5.8 (1.0.6) freezes after an episode has been deleted, but only trying to leave the current view. This has been seen when run from two entirely separate instances of the player, on Fedora 23 and FireTV. My Plex Media Server is currently at 0.9.16.4, but has undergone a few updates over the past few months, with the same crashing/freezing behavior from the new PMP. 

PMS 0.9.16.4 - Ubuntu 15.10 [4.2.0-35-generic x86_64]
PMP 2.5.8 (build 1.0.6) - Fedora 23 (Workstation Edition) [4.4.6-301.fc23.x86_64]
PMP 2.5.8 (build 1.0.6) - Amazon Fire TV  (Settings state PMP it is version 4.24.1.561)

The PMS log offers nothing other than the INFO of the library update (due to the change created by deleting an episode). The PMP log does offer some insight with the following:

```
2016-04-14 15:44:29 [ INFO  ] JS: [Commands] Executing deleteItem
2016-04-14 15:44:29 [ INFO  ] JS: [Commands] Executing deleteFile
2016-04-14 15:44:30 [ INFO  ] JS: Reusing current background image
2016-04-14 15:44:57 [ DEBUG ] InputComponent.cpp @ 86 - Input received: source: "Keyboard" keycode: "Backspace" 
2016-04-14 15:44:59 [ INFO  ] JS: Updater: could not get update list from plex.tv 
2016-04-14 15:44:59 [ INFO  ] JS: Updater: Checking again in 360 minutes
```

It would seem there is an attempt to gather more information from plex.tv, which is where it hangs. Only resolution is killing PMP.

To reiterate, the steps to create the freeze are as follows:
Delete an Episode, success.
Interface takes me back to the season list, which has been updated to show deleted file.
Hit Escape or Back so as to leave that season view and...
Waiting animation loops forever in that season view.

**Comment by firasdib** [comment]
Can you give me the full PMP logs?

**Comment by wm4** [comment]
Forum thread: https://forums.plex.tv/discussion/215628/deleting-content-from-pmp-1-0-6-crashes-player (no new information yet).

**Comment by cedell** [comment]
Sure, here is a full one. FYI, I went in and `<snipped>
[plexmediaplayer.log.txt](https://github.com/plexinc/plex-media-player/files/226097/plexmediaplayer.log.txt)

` out my clientID and external IP.

**Comment by cedell** [comment]
As another follow up, I have now realized that hitting backspace a couple of time during the loading animation allows me to jump back out of that view. This works for both the FireTV and the PHT build under Fedora. I feel like the logs are not properly capturing the freeze event, as the entry for "could not get update list from plex.tv " occurs prior to the action. Any thoughts?

**Comment by firasdib** [comment]
@cedell I just tried to reproduce this and was able to delete a movie just fine. Could you please give me more details?

- Did you have any filters on?
- Was this the last media in the library?
- Any other settings you can think of?

Can you please give me the logs for PMS when you tried to delete something?

**Comment by cedell** [comment]
Sure, here are logs from right now (from Ubuntu server and Fedora client). As for your questions:

- Not sure what you mean by filters. Do you mean at the firewall level? If so, no. I tried it with my firewall on and then off, with the same results.
- No, it happens for all content regardless of when it was added.
- The server is pretty close to stock configuration with the exception of DLNA and Remote Access being turned off. I have the trakt.tv plugins, which I removed in case it was the issue, but that didn't change anything either. Something that caught my eye in this last log was the"JS: Server without `friendlyName` skipped." in the PMP log. Not sure if that's relevant or not.
- Each of my test were with single episodes. I also tested it with a movie, with the same results.

What I have realized now, as I mentioned in my previous post, is that hitting the backspace again actually kills the loop. Instead of continuing to travel backwards, the waiting animation will actually go away, allowing me to hit backspace a third time to escape that view. It's still an issue, clearly, but at least I now know that the player isn't completely frozen.

[plexmediaplayer.log.txt](https://github.com/plexinc/plex-media-player/files/237760/plexmediaplayer.log.txt)
[Plex Media Server.log.txt](https://github.com/plexinc/plex-media-player/files/237761/Plex.Media.Server.log.txt)

**Comment by firasdib** [comment]
By filters I meant those in the app, like showing only unwatched movies etc.

Just to re-iterate and verify: the items you deleted, were they the last item in that library/view? For example, the last episode of a season/series?

**Comment by cedell** [comment]
Ah okay. No, I've never even tried filtering. Also, if it helps, for the route into a show I always choose "TV Shows" and then "Browse All."

As for the episode, it doesn't matter which one it is in the season. Typically I am deleting the lowest episode number, as I delete every episode after watch content in order. But, in my tests I have also tried the last episode, with identical results. And, remember that this occurs with movies as well. Essentially it is all of my media.

Here's the PMS with verbose/debug on. Sorry, it didn't occur to me before. The delete occurs at 10:06:46, I waited a few and tried to hit the backspace the first time at 10:06:48. Hope that helps.

[Plex Media Server.log.txt](https://github.com/plexinc/plex-media-player/files/238778/Plex.Media.Server.log.txt)
